libs/dal/src/repositories/subscriber/subscriber.repository.ts

```diff
@@ -32,6 +33,68 @@ export class SubscriberRepository extends BaseRepository<SubscriberDBModel, Subs
     );
   }
 
+  async bulkCreateSubscribers(subscribers, environmentId: EnvironmentId, organizationId: OrganizationId) {
+    const bulkWriteOps = subscribers.map((subscriber) => {
+      const { subscriberId, ...rest } = subscriber;
+
+      return {
+        updateOne: {
+          filter: { subscriberId, _environmentId: environmentId, _organizationId: organizationId },
+          update: { $set: rest },
+          upsert: true,
+        },
+      };
+    });
+
+    let result;
+    try {
+      const response = await this.bulkWrite(bulkWriteOps);
+      result = response.result;
+    } catch (e) {
+      if (!e.writeErrors) {
+        throw new DalException(e.message);
+      }
```

Comment:
If the error is not from write errors related to the bulk execution, throw the error. 
Otherwise, we want to parse that into the response and not throw error